src/server.cpp

```diff
@@ -127,6 +127,109 @@ void *ServerThread::run()
 	return nullptr;
 }
 
+class MapThread : public Thread
+{
+public:
+
+	MapThread(Server *server):
+		Thread("Server"),
+		m_server(server)
+	{}
+
+	void *run();
+
+private:
+	Server *m_server;
+};
+
+void *MapThread::run()
+{
+	BEGIN_DEBUG_EXCEPTION_HANDLER
+
+	const float save_interval =
+		g_settings->getFloat("server_map_save_interval");
+	const float map_timer_and_unload_dtime = 2.92f;
+
+	float savemap_timer = 0.0f;
+	IntervalLimiter map_timer_and_unload_interval;
+
+	u64 last_time = porting::getTimeMs();
+	while (!stopRequested()) {
+		sleep_ms(100); // no busy loop
+
+		u64 curr_time = porting::getTimeMs();
+		float dtime = (curr_time - last_time) / 1000.0f;
+		last_time = curr_time;
+
+		ServerEnvironment &env = m_server->getEnv();
+		ServerMap &map = (ServerMap&) env.getMap();
+
+		// Save map, players and auth stuff
+		CompressedBlockList compressed_blocks;
```

Comment:
Not needed yet, could be moved after `continue;` for a cleaner structure.

Reply:
+1